backend/app/Http/Controllers/AuthController.php

```diff
@@ -68,4 +78,64 @@ public function update(UpdatePasswordRequest $request, User $user)
 
         return UserResource::make(['message' => 'Password updated successfully.']);
     }
+
+    public function forgotPassword(ForgotPasswordRequest $request)
+    {
+        $userEmail = $request->validated();
+
+        $status = Password::sendResetLink(
+            ['email' => $userEmail['email']]
+        );
+
+        if ($status === Password::RESET_LINK_SENT) {
+            return ForgotPasswordResource::make([
+                'message' => 'The link was sent, please check your email',
+            ]);
+        } else {
+            return ForgotPasswordResource::make([
+                'message' => 'Link was already sent, please check your email',
+            ]);
+        }
+    }
+
+
+    public function resetPassword(ResetPasswordRequest $request)
+    {
+        // Reset the user's password
+        $validated = $request->validated();
+
+        $user = User::where('email', $validated['email'])->first();
+
+        if (!$user || Hash::check($validated['password'], $user->password)) {
+            throw ValidationException::withMessages([
+                'message' => 'password should not be an old password',
+            ]);
+        }
+
+        $status = Password::reset(
+            $request->only('email', 'password', 'token'),
```

Comment:
```suggestion
            $->only('email', 'password', 'token'),
```
use already extracted input data